### PR TITLE
Testing guides

### PR DESCRIPTION
**Pull request by madox2**

### CODE REVIEW

File: testing/README.md
Comment (IvanaKra) [inline testing/README.md:1]: I would change the last sentence to make it more clear: All existing guidelines in this repo apply equally to testing-related code.
I would also appreciate some examples of situations when testing is needed and when not.
What about balance between manual and automated testing? Are there any recommendations?
What about long lasting projects, should we do maintenance of tests?
Add Role responsibilities: developer : testing his code (unit and manual), code reviewer: testing (manual)

Comment (jansvantner) [reply testing/README.md:1]: @IvanaKra please use code suggestion

File: testing/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Testing Guidelines
+
+## General
+
+- Code coverage is easy to measure but real test quality is more complicated than that, write test where it matters
+- Writing unit tests is automatically considered to be a part of the developer's job
+- Manual feature testing is executed by the developer and should be also performed by the code reviewer
+- Other guidelines in this repo also apply to the code produced during testing
```
Comment (jansvantner) [inline testing/README.md]: add:
```
- Developers are responsible for testing as well. Especially when there are no dedicated testers on the project. 
- Use four-eyes principle to enhance quality of the testing
- Use LEAN when writing the tests - write tests with highest customer value first (e.g. testing high-risk code, highly reused code or often used functionality) 
```

Comment (madox2) [reply testing/README.md]: > Use four-eyes principle to enhance quality of the testing

co tym konkretne myslis?

> Use LEAN when writing the tests

asi viem co chces povedat, ale LEAN je viac ako to

File: testing/README.md
Comment (IvanaKra) [reply testing/README.md:1]: Bod 1 - zapracujeme
Bod 2 - pridame ze zalezi od situacie
Bod 3 - rovnako ako bod 2
Bod 4 - robim maitenance testov automaticky, ked prestanu testy fungovat (you break it you fix it strategy)
testy bezia pocas CI 
Bod 5 - nic

File: testing/README.md (hunk 1 above)
Comment (IvanaKra) [reply testing/README.md]: Update riadok 7 Manual feature testing must be executed...
Testovat by to mala ina osoba ako ten co developuje.
Use four-eyes principle to enhance quality of the  feature testing (one develops and test, another should test the feature)
vyhodit LEAN

**Pull request merged**